File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/abstract_factory/lab/conventional_garden.cpp
#include "conventional_garden.h"
conventional_garden::conventional_garden(conventional_lettuce* a_lettuce_prototype)
    :garden(a_lettuce_prototype)
{
}

void conventional_garden::set_lettuce_prototype(conventional_lettuce* a_lettuce_prototype)
{
    garden::set_lettuce_prototype(a_lettuce_prototype);
}

conventional_tomato* conventional_garden::make_tomato() const
{
	return new conventional_tomato(3, 1);
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/abstract_factory/solution/ConventionalGarden.java
// ConventionalGarden.java

public class ConventionalGarden extends Garden
{
	public ConventionalGarden(ConventionalLettuce aLettucePrototype)
	{
		super(aLettucePrototype);
	}

	public void setLettucePrototype(ConventionalLettuce aLettucePrototype)
	{
		super.setLettucePrototype(aLettucePrototype);
	}

	public ConventionalTomato makeTomato()
	{
		return new ConventionalTomato(1, 2);
	}
}
